matplotcheck/raster.py

```diff
@@ -114,19 +112,65 @@ def assert_legend_accuracy_classified_image(
         Finally those two arrays of strings are compared. Passes if they match.
         """
         # Retrieve image array
-        im_data = []
-        if self.ax.get_images():
-            im = self.ax.get_images()[0]
-            im_data, im_cmap = im.get_array(), im.get_cmap()
+        im_data = self.get_plot_image()
+        im = self.ax.get_images()[0]
+        im_cmap = im.get_cmap()
+
         assert list(im_data), "No Image Displayed"
 
+        # Retrieve legend entries and find which element of all_label_options
+        # matches that entry
+        legend_dict = self.get_legend_labels(all_label_options)
+
+        # Create two copies of image array, one filled with the plot data class
+        # labels (im_data_labels) and the other with the expected labels
+        # (im_expected_labels)
+        im_class_dict = {}
+
+        # Accounting for images that don't start at 0
+        correction = np.min(im_data)
+
+        for val in np.unique(im_data):
+            im_class_dict[val - correction] = legend_dict[
+                im_cmap(self.ax.get_images()[0].norm(val))
+            ]
+        im_data_labels = [
+            [im_class_dict[val - correction] for val in row]
+            for row in im_data.data
+        ]
+        im_expected_labels = [
+            [all_label_options[val - correction][0] for val in row]
+            for row in im_expected
+        ]
+
+        # Check that expected and actual labels match up
+        assert np.array_equal(
+            im_data_labels, im_expected_labels
+        ), "Provided legend labels don't match labels found."
+
+        # IMAGE TESTS/HELPER FUNCTIONS
+
+    def get_legend_labels(self, all_label_options):
```

Comment:
@nkorinek this helper is confusing. i expected it to return the labels that it found in the existing legend. But it seems to check the labels? It seems to me having a helper that just returned the labels would be really helpful. then you can have a separate helper that checks that the correct words are in each label element. does that make sense? this function name suggests it will just grab the legend labels which is ideal. i may be confused by what is happening here... so we can chat i fyou want

Reply:
<img width="1154" alt="Screen Shot 2020-04-20 at 5 48 51 PM" src="https://user-images.githubusercontent.com/7649194/79810020-3be84200-832f-11ea-85de-3b7206f058f4.png">
i'm getting some odd outputs from this method. does this make sense to you?

Reply:
i am guessing that could be rgb values and then data value exept the 4th value is always 1.0.

Reply:
```
def assert_legend_labels(expected_labels):
     self.get_legend_labels()
     # check stuff
     # Make sure that the expected labels are in the legend labels
```